### PR TITLE
RNMobile Add size options to mobile  image block

### PR DESCRIPTION
**Pull request by cameronvoell**
## Description

This PR covers adding image size options to the gutenberg mobile editor as specified in [gb-mobile issue 1263](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1329)
## How has this been tested?

Tested using iPhone X simulator running iOS 12.4 and Samsung S10 running API 28. Steps used for testing:
1.  Building Example app with metro bundler running with [gutenberg-mobile master branch 0e597a4](https://github.com/wordpress-mobile/gutenberg-mobile/commit/0e597a4afb8c7dfbcb4da0ca943d73102941f52b)
2. Click on an Image block (that already has an image)
3. Press the gear button to bring up Image Options menu
4. Verify new Size option exists, click size options
5. Verify size options are shown and choose one.
6. Re-open Image options menu and verify previously selected size is now shown as selected.

For verifying changes are correctly submitted to web backend:
1. Run the metro bundler for this branch running in gutenberg-mobile and build either Android or iOS WordPress app
2. Use a wordpress.com account and navigate to a post, and begin editing in Gutenberg editor.
3. Follow steps above to update Image Size options. Click Update.
4. One the web, log in to the wordpress.com account, navigate to the article, and verify that the new Image size setting is shown.

## Screenshots 
size-options

### Utilizes ActionSheet on iOS:
image

## Types of changes

This code introduces the ability to update Image Size Options from the Gutenberg editor, the same way you can in the Aztec Editor, and in Gutenberg on the web. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by etoledom** [comment]
@cameronvoell - Thanks for your pull request! 🎉 

I saw the failing tests and noticed this error message:

`packages/block-library/src/image/edit.native.js: Support for the experimental syntax 'classProperties' isn't currently enabled (60:24):`

```
  58 | 
  59 | class ImageEdit extends React.Component {
> 60 | 	menuTransitionTimeout = 0;
     | 	                      ^
  61 | 
```

Could you please check that out? Thank you!

We are temporarily using the `rnmobile/master` branch as our master for mobile development. It would be nice if you could update from and target that branch.

**Comment by etoledom** [comment]
I gave it a run on iOS and this happened:

The first time it works, but the second time, the native bottom sheet dismisses automatically and the whole app became unresponsive.

Please let me know if you can reproduce this issue or I might be doing something wrong!

image-sizes

**Comment by cameronvoell** [comment]
> I gave it a run on iOS and this happened:
> 
> The first time it works, but the second time, the native bottom sheet dismisses automatically and the whole app became unresponsive.
> 
> Please let me know if you can reproduce this issue or I might be doing something wrong!

Yea it looks like there is some issue if there is not large enough of a gap between when the BottomSheet is dismissed and when the ActionSheetIOS is shown. I think once [BOTTOM_SHEET_TO_PICKER_ANIMATION_DELAY_MILLISECONDS](https://github.com/WordPress/gutenberg/blob/2533fb10461c4ec321d842cd680a0f61b8097dae/packages/block-library/src/image/edit.native.js#L338) gets large enough, the issue can not be reproduced, but unfortunately the minimum might be hardware specific. 

I'll take a look to see if I can get more details about what is going on, especially around that unresponsive state. And if there continues to be issues with the transition from BottomSheet to ActionSheetIOS, maybe we can use a BottomSheet for selecting Sizes options as well, at least for the first pass.

**Comment by cameronvoell** [comment]
> @cameronvoell - Thanks for your pull request! 🎉
> 
> I saw the failing tests and noticed this error message:
> 
> `packages/block-library/src/image/edit.native.js: Support for the experimental syntax 'classProperties' isn't currently enabled (60:24):`
> 
> ```
>   58 | 
>   59 | class ImageEdit extends React.Component {
> > 60 | 	menuTransitionTimeout = 0;
>      | 	                      ^
>   61 | 
> ```
> 
> Could you please check that out? Thank you!
> 
> We are temporarily using the `rnmobile/master` branch as our master for mobile development. It would be nice if you could update from and target that branch.
> 
> Also creating a branch on `gutenberg-mobile` to test this one is always helpful! From there you can create a Draft PR if needed. 😃

Thanks for the feedback, I'll target rnmobile/master, create a gb-mobile branch, and check out that failing test error :+1:

**Comment by cameronvoell** [comment]
> Please let me know if you can reproduce this issue or I might be doing something wrong!

Fixed this using a regular [Modal onDismiss prop](https://facebook.github.io/react-native/docs/modal#ondismiss) that I had missed previously instead of trying to time showing the Picker with the BottomSheet close animation. 

The unresponsive app looks like it's related to an issue where modals remain covering the UI if you try and show Alert, Modal, and looks like ActionSheetIOS also, before the modal is dismissed (more info [here](https://www.bountysource.com/issues/38533054-ios-ui-will-be-blocked-when-show-alert-while-closing-modal)). We shouldn't have to worry about that now that we're using the onDismiss prop though. To test I tried the lower end emulators while debugging which previously showed the issue, but now it looks good. 

>I saw the failing tests and noticed this error message:
>
> packages/block-library/src/image/edit.native.js: Support for the experimental syntax 'classProperties' isn't currently enabled (60:24):

This change should clear up that "experimental syntax 'classProperties'" test fail also :-)

**Comment by cameronvoell** [comment]
> Also creating a branch on `gutenberg-mobile` to test this one is always helpful! From there you can create a Draft PR if needed. 😃

Looks like I don't have access for making branches on wordpress-mobile/gutenberg-mobile. But I made a branch on my gutenberg-mobile fork here: [add/image-size-options](https://github.com/cameronvoell/gutenberg-mobile/tree/add/image-size-options). Github username cameronvoell in case you can add me to gutenberg-mobile :-) Otherwise I'll just be careful about keeping my fork up to date, which I'm used to doing with gutenberg anyway.

**Comment by iamthomasbishop** [comment]
@cameronvoell looks pretty solid overall!  Two small design requests.

1.  Can we left align the text labels on the bottom sheet (on Android specifically) that houses the image size options? I believe on the 16px keyline, iirc.
2. Are we able to add a top description to the action sheet on iOS? If so, that title would read `Image Size`.

**Comment by cameronvoell** [comment]
@iamthomasbishop 
> * Can we left align the text labels on the bottom sheet (on Android specifically) that houses the image size options? I believe on the 16px keyline, iirc.

16px, let me know if you want to update (sorry don't have good visual instincts for this yet!):
Screenshot_20190829-192052_Gutenberg

> * Are we able to add a top description to the action sheet on iOS? If so, that title would read `Image Size`.

image

**Comment by etoledom** [comment]
Thanks for the updates @cameronvoell !

I did a fast test on iOS and I saw this error (or warning on red?) on the JS console when the app loads:

```
Warning: Failed prop type: Invalid prop `onModalHide` of type `boolean` supplied to `ReactNativeModal`, expected `function`.
    in ReactNativeModal (at bottom-sheet/index.native.js:126)
    in BottomSheet (at dark-mode/index.native.js:50)
    in _class (at edit.native.js:255)
    in RCTView (at View.js:35)
    in View (at edit.native.js:368)
    in TouchableWithoutFeedback (at edit.native.js:362)
    in MediaUpload (at edit.native.js:450)
    in ImageEdit (at dark-mode/index.native.js:50)
    in _class (at edit.native.js:18)
```

And then this warning each time the BottomSheet dismisses:
```
Possible Unhandled Promise Rejection (id: 0):
TypeError: _this.props.onModalHide is not a function
TypeError: _this.props.onModalHide is not a function
```

They are clearly related. I didn't see this happening on Android.

----

As I recall from the last review, the Settings bottom sheet was presenting itself again after the Sizes picker was dismissed? I might be wrong though. Now when the Sizes picker is dismissed, there is no Image Settings anymore. I don't think this is a blocker and could be OK as a first iteration anyway. But I'd like to check with @iamthomasbishop what's the expected behavior.

Here is a gif with the current behavior:

image_sizes_dismiss


----

Code-wise looks good to me!
We have a `BottomSheetPickerCell` that does pretty much what we need here. I wonder if it can be used instead of a regular Cell. It seems like we are not using it anywhere though.

----

I'm a bit concerned about the behavior of the setting. Currently we are modifying the sizeSlug and saving it on the block attributes. This works well, and if this is the extend of this PR there's nothing else to add ✅ 

But this will not actually modify the size of the image displayed when the post is published. The idea of the slug is to select the properly sized image URL. And this information is fetched from remote. 

This is where the image info is fetched on the web implementation:
https://github.com/WordPress/gutenberg/blob/727e96282eec0aeabb62418bc224577d89418a03/packages/block-library/src/image/edit.js#L944

This is where the url is selected based on the slug:
https://github.com/WordPress/gutenberg/blob/727e96282eec0aeabb62418bc224577d89418a03/packages/block-library/src/image/edit.js#L500

And related to this, the size options are dynamic, so we should not hardcode them. I'm not sure if this is also fetched from remote, it might be 🤔 

https://github.com/WordPress/gutenberg/blob/727e96282eec0aeabb62418bc224577d89418a03/packages/block-library/src/image/edit.js#L938


Sadly we don't have networking on the native side of the project (as far as I know), so we can't implement this in the same way yet, and I don't think that adding networking should be part of this task.

For me it would be fine to keep this PR as it is on this regard, and hide this option under a `dev` flag for the mean time. I'd love to hear @hypest 's opinion on this.

**Comment by cameronvoell** [comment]
> I did a fast test on iOS and I saw this error (or warning on red?) on the JS console when the app loads:
> 
> ```
> Warning: Failed prop type: Invalid prop `onModalHide` of type `boolean` supplied to `ReactNativeModal`, expected `function`.
> ```

Sorry missed that warning, fixed here: https://github.com/WordPress/gutenberg/pull/17245/commits/b9aecf5529c43bed33d8d23afe44f904ff9bd243

**Comment by iamthomasbishop** [comment]
> As I recall from the last review, the Settings bottom sheet was presenting itself again after the Sizes picker was dismissed? I might be wrong though. Now when the Sizes picker is dismissed, there is no Image Settings anymore. I don't think this is a blocker and could be OK as a first iteration anyway. But I'd like to check with @iamthomasbishop what's the expected behavior.

I think ideally, we'd bring the Image Settings sheet back, but I don't necessarily think it's a blocker for the first iteration.

**Comment by iamthomasbishop** [comment]
@cameronvoell The description on the iOS ActionSheet looks good 👍 After taking a look at the Android bottom sheet, it looks like what you've currently got is about 28px inset. While I was proposing 16px, I tried mocking it up in Figma and 16px looks a little tight. Can we try aligning it to the 24px keyline (from the left edge) and ship it?

image

I believe all of the existing sheets use the 16px inset, but in this type of case where there aren't icons, 24px breathes a little more.

File: packages/block-library/src/image/edit.native.js
Comment (Tug) [inline packages/block-library/src/image/edit.native.js]: Having so many functions defined in the render method isn't good practice. This could probably be moved outside the class or as a class method if we get the `sizeSlugValue` from props.

File: packages/block-library/src/image/edit.native.js
Comment (Tug) [inline packages/block-library/src/image/edit.native.js]: This could be simplified if we instead define a mapping (also probably outside render):
```js
const sizeOptionLabels= {
	[ IMAGE_SIZE_THUMBNAIL ]: __( 'Thumbnail' ),
	[ IMAGE_SIZE_MEDIUM ]: __( 'Medium' ),
	[ IMAGE_SIZE_LARGE ]: __( 'Large' ),
	[ IMAGE_SIZE_FULL_SIZE ]: __( 'Full Size' ),
}
```
Then you could simply access it with `sizeOptionLabels[ sizeSlugValue || DEFAULT_SIZE_SLUG ]` instead of using `getSizeSlugDisplay`.

File: packages/block-library/src/image/edit.native.js
Comment (Tug) [inline packages/block-library/src/image/edit.native.js]: probably worth having it as a bound class method instead

File: packages/block-library/src/image/edit.native.js
Comment (Tug) [inline packages/block-library/src/image/edit.native.js]: This ImageEdit component is getting pretty large, do you think it's worth moving the size picker to its own component? We could move the `IMAGE_SIZE_*` consts there as well, what do you think?

File: packages/components/src/mobile/bottom-sheet/index.native.js
Comment (Tug) [inline packages/components/src/mobile/bottom-sheet/index.native.js]: I'm not sure I understand how `props.onDismiss` is different from `props.onClose` here, could you expand on this?

File: packages/components/src/mobile/picker/index.ios.js
Comment (Tug) [inline packages/components/src/mobile/picker/index.ios.js]: Nice addition there 👍

**Review by Tug — COMMENTED** [review]
I had a few comments on the code that I think we should address before considering adding support for persisting the size option. We'll actually probably want to do this in another PR.
Do you mind hiding the size selection option behind a feature flag?

File: packages/block-library/src/image/edit.native.js
Comment (etoledom) [inline packages/block-library/src/image/edit.native.js]: This style seems to have been removed [here](https://github.com/WordPress/gutenberg/pull/17245/commits/ef79f0852f250b6e04d3ba7d95deb65d4d47fc05).
Is this still needed?

File: packages/block-library/src/image/edit.native.js
Comment (Tug) [inline packages/block-library/src/image/edit.native.js]: `<Picker ... leftAlign />` is valid as well

File: packages/components/src/mobile/bottom-sheet/index.native.js
Comment (cameronvoell) [reply packages/components/src/mobile/bottom-sheet/index.native.js]: `props.onDismiss` is used for calling functions to occur **after** the Modal is done dismissing. We use this to check if we should open another modal for Image size settings. 

`props.onClose` is assigned the function that will perform the action that closes the modal. So in image editor.native.js for example, we assign a `onImageSettingsClose` function that sets state.showSettings to false.

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: Thanks for catching that, removed here: ad81e1596f6b625a5df7641aa5d48591a278768d

File: packages/components/src/mobile/picker/index.ios.js
Comment (cameronvoell) [reply packages/components/src/mobile/picker/index.ios.js]: Thanks, glad the code accommodated thomas's request here very easily

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: I agree, I'll try for a simple refactor commit now that moves the size picker to its own component, and addresses some of the other good ideas around code you had in other comments. I'll keep the __DEV__ flag in mind as well in case minor changes will help make incorporating that nice and clean.

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: Moved to ImageSizePicker in 7983f04c71df03900122d95e885749c6eb1665c2

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: Agreed this is much cleaner. Added to ImageSizePicker in 7983f04c71df03900122d95e885749c6eb1665c2

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: Refactored to new ImageSizePicker component here: 7983f04c71df03900122d95e885749c6eb1665c2

**Comment by cameronvoell** [comment]
> @cameronvoell After taking a look at the Android bottom sheet, it looks like what you've currently got is about 28px inset. While I was proposing 16px, I tried mocking it up in Figma and 16px looks a little tight. Can we try aligning it to the 24px keyline (from the left edge) and ship it?

Updated in 3109e1136b7bf72f89ea2db1e48eca86e01b057e , See Image:
Screenshot_20190903-170551_Photos

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: I moved this code to ImageOptionsPicker, but have not changed it to a bound class method yet. I'm using [this pattern from MediaUpload](https://github.com/WordPress/gutenberg/blob/a400f7b354a6f5dc3dd59daef3d4a90069dff54b/packages/block-editor/src/components/media-upload/index.native.js#L91-L103) for passing the presentPicker function to the container of our component. 
(github insert code kept deformating this code)
image

If we simply move onPickerPresent to be a class method, we'll no longer have a reference to picker that is assigned in the Picker props to the picker instance. 
The current code feels like an antipattern to me, but I figured consistency was a quick, safe option, and that maybe you or someone else on the team would have feedback on how we could update this code in MediaUpload and here at once. If the cleanest way to do this requires a little refactoring of the Picker, perhaps it is not the best time to do this though since the same pattern is used elsewhere in the code.

**Comment by cameronvoell** [comment]
> Do you mind hiding the size selection option behind a feature flag?

Size selection is now hidden behind `__DEV__` flag as we discussed in chat. See: ce2af941a8014be33f80233bb97f5201709d0f2a

gutenberg-mobile branch caught up with develop containing an updated reference to my gutenberg branch here: [add/image-size-options](https://github.com/cameronvoell/gutenberg-mobile/tree/add/image-size-options)

File: packages/block-library/src/image/edit.native.js
Comment (etoledom) [inline packages/block-library/src/image/edit.native.js]: Just curious:
Where you able to test `BottomSheet.PickerCell` component instead of the regular `Cell`?

**Comment by etoledom** [comment]
I did a run test and it seems to work pretty well in both iOS and Android 🎉

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: I did not test BottomSheet.PickerCell yet, but it looks like it would need to be adjusted to avoid the iOS issue where it freezes if an ActionSheet is shown before the initial Modal is dismissed. Right now it looks like it would try and show the picker right when the previous cell is clicked:  
https://github.com/WordPress/gutenberg/blob/5774046c216a96d95fcbcfd2903b74f5f69a0611/packages/components/src/mobile/bottom-sheet/picker-cell.native.js#L16-L25

I'll look into abstracting the show picker on dismiss functionality into PickerCell, since it seems this might be a good pattern to reuse elsewhere.

**Comment by Tug** [comment]
> We have a BottomSheetPickerCell that does pretty much what we need here. I wonder if it can be used instead of a regular Cell. It seems like we are not using it anywhere though.

👍 we should be reusing this, and if limited add features to it instead

File: packages/block-library/src/image/edit.native.js
Comment (Tug) [reply packages/block-library/src/image/edit.native.js]: I don't think this is a pattern we should try to reproduce here. It doesn't follow the SOC principle and makes the code harder to read imo. I agree MediaUpload is confusing in its current stage though.

File: packages/block-library/src/image/image-size-picker.native.js
Comment (Tug) [inline packages/block-library/src/image/image-size-picker.native.js]: we could use `map( sizeOptionLabels, ( option, label ) => ( { value: option, label } ) )` instead here

File: packages/block-library/src/image/image-size-picker.native.js
Comment (Tug) [inline packages/block-library/src/image/image-size-picker.native.js]: Still `leftAlign={ true }` => `leftAlign`. It's simpler

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: Sorry misunderstood you here earlier. I saw the second comment, will update in the next commit.

File: packages/block-library/src/image/image-size-picker.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/image-size-picker.native.js]: Sorry misunderstood your comment last time, fixed 717d27be68f3073720792815a3f6ae0418812aee

File: packages/block-library/src/image/image-size-picker.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/image-size-picker.native.js]: Thanks for the tip. Added here: 717d27be68f3073720792815a3f6ae0418812aee
 I didn't realize you could rely on object key orderings, great to know 👍  https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/

**Comment by cameronvoell** [comment]
> Just curious:
> Where you able to test `BottomSheet.PickerCell` component instead of the regular `Cell`?

> 👍 we should be reusing this, and if limited add features to it instead
> 
> I think it would make more sense to use it in-place and to make it so that picking an item shows the picked item in the main BottomSheet rather than closing everything

Utilizing the BottomSheet.PickerCell here ff0c5dc93d45077b3ca45e883c2d6e57d14570a1.

>  [This is the comment I was following for the transition from first to second sheet.](https://hogwartsp2.wordpress.com/2019/08/19/camerons-trial-gutenberg-mobile-image-options-part-ii/#comment-14435)

Just to explain my thought process, the link above is the p2 post when Thomas told me explicitly 

> "My one concern initially is that the transition feels natural. Rather than just “stacking” the modals on top of each other, I am imagining the initial sheet would dismiss and – with a small delay – the incoming sheet would slide in."

 That is why I went strong in the direction of getting to the picker to show after the BottomSheet was dismissed initially. Also, the BottomSheet.PickerCell was not used anywhere in the code base which is one reason why I didn't find it initially. All that being said, the code is way cleaner utilizing the BottomSheet.PickerCell. For triggering the PickerCell after the initial BottomSheet is dismissed, I need to think a little more.

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: See comment https://github.com/WordPress/gutenberg/pull/17245#issuecomment-528232485

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: Utilizing `BottomSheet.PickerCell` allowed me to get rid of this code, but unfortunately, `PickerCell` is using the same pattern. I think all of this calling child methods from parent are stemming from `presentPicker()` method in the [ios ](https://github.com/WordPress/gutenberg/blob/8dc1de9ea27e43a3c8b32a6e23f5548ae9c552b3/packages/components/src/mobile/picker/index.ios.js#L13) and [android](https://github.com/WordPress/gutenberg/blob/8dc1de9ea27e43a3c8b32a6e23f5548ae9c552b3/packages/components/src/mobile/picker/index.android.js#L29) picker classes. I didn't have immediate ideas on how to refactor that, but have been doing some reading. Would be great to update so we stop breaking the SOC principle like you mentioned.

File: packages/components/src/mobile/bottom-sheet/index.native.js
Comment (cameronvoell) [inline packages/components/src/mobile/bottom-sheet/index.native.js]: Think we can safely leave these props here now. They will be useful as a reference in the future in case we want to call a method once the BottomSheet is closed.

File: packages/block-library/src/image/edit.native.js
Comment (Tug) [inline packages/block-library/src/image/edit.native.js]: Nitpicking here, but I'm thinking we should probably have done the opposite:
```
const sizeOptions = [
	{ value: IMAGE_SIZE_THUMBNAIL, label: __( 'Thumbnail' ) },
	{ value: IMAGE_SIZE_MEDIUM, label: __( 'Medium' ) },
	{ value: IMAGE_SIZE_LARGE, label: __( 'Large' ) },
	{ value: IMAGE_SIZE_FULL_SIZE, label: __( 'Full Size' ) },
];
const sizeOptionLabels = sizeOptions.reduce( ( result, { value, label } ) => { result[value] = label; return result; }, {} );
```
Or rather, since this is only called whenever the value of the picker changes, I don't think precomputing `sizeOptionLabels` is even worth it:
``` 
<BottomSheet.PickerCell
    value={ sizeOptions.find( { value: sizeSlug || DEFAULT_SIZE_SLUG } ) }
    ...
/>
```

**Review by Tug — CHANGES_REQUESTED** [review]
Looks really good to me now 👍 

One thing I noticed is that dismissing the size picker by swiping down dismisses both modals. I think we should find a way to fix that before we get it merged

**Edit: I saw Thomas gave you his [approval](https://github.com/WordPress/gutenberg/pull/17245#issuecomment-526660634) for that part so nevermind**

**Comment by iamthomasbishop** [comment]
@cameronvoell sorry for the delay on the last ping – screenshot above looks great.

**Comment by etoledom** [comment]
Hey @cameronvoell - Thank you for the update and for the effort!

I noticed a regression on Android where the cells are not aligned left anymore:

Screenshot_20190905-180827

**Comment by cameronvoell** [comment]
> Hey @cameronvoell - Thank you for the update and for the effort!
> 
> I noticed a regression on Android where the cells are not aligned left anymore:
> 
> Screenshot_20190905-180827

Yea I'll re-add the styling code to BottomSheet.PickerCell.

File: packages/block-library/src/image/edit.native.js
Comment (cameronvoell) [reply packages/block-library/src/image/edit.native.js]: I appreciate the nitpicking. I need to work on considering lodash and Array functions in these cases. Still transitioning from Java/Kotlin world :-P.  Let me know if you want me to update this.

Comment (gziolo) [reply packages/block-library/src/image/edit.native.js]: Aren't all those constants shared between web and mobile? I would expect they exist already and could be reused.

Comment (etoledom) [reply packages/block-library/src/image/edit.native.js]: In this case, `imageSize` comes from block-editor settings: 
https://github.com/WordPress/gutenberg/blob/de0cc66a4f2545ead0ddef544d434349f371078c/packages/block-library/src/image/edit.js#L938

https://github.com/WordPress/gutenberg/blob/de0cc66a4f2545ead0ddef544d434349f371078c/packages/block-library/src/image/edit.js#L554-L555

I did a small test and we have this available on mobile too. So we can reuse it 👍

**Review by etoledom — COMMENTED** [review]
I think it looks good. There are still some details to be polished but since we are not yet shipping it, we can continue workin on a separated PR.

One detail to note:

From https://github.com/WordPress/gutenberg/pull/17245#issuecomment-528232485
> That is why I went strong in the direction of getting to the picker to show after the BottomSheet was dismissed initially.

I think this is the right way to go too. And probably this was just a misunderstanding of @Tug's comment [here](https://github.com/WordPress/gutenberg/pull/17245#issuecomment-528074708):

> I think it would make more sense to use it in-place and to make it so that picking an item shows the picked item in the main BottomSheet rather than closing everything

He can confirm, but I think he meant to bring back the main settings sheet after the Sizes sheet was dismissed (going according with @iamthomasbishop plan). But Thomas then said it was fine to not bring it back for now, so everything was fine 👍 

There is a small lint error to be fixed before merging though. After that, I'd say let's :shipit: and move forward. We can continue improve it on a new PR, including what was mentioned [here](https://github.com/WordPress/gutenberg/pull/17245/files#r334790181)

**Comment by cameronvoell** [comment]
> There is a small lint error to be fixed before merging though. After that, I'd say let's :shipit: and move forward. 

Fixed! https://github.com/WordPress/gutenberg/pull/17245/commits/c2d4fe5b02ada0fb2345400737dd52994c3ae021

> We can continue improve it on a new PR, including what was mentioned [here](https://github.com/WordPress/gutenberg/pull/17245/files#r334790181)

^ Added to new issue here, for tracking Image Options improvements: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1452

**Review by etoledom — APPROVED** [review]
Thank you @cameronvoell , great job! 🎉

**Comment by etoledom** [comment]
@gziolo could you please help us mergin? Since @Tug has the `requested changes` status, I can't press the merge button.

His feedback has been taken into account and all the remaining feedback will be handled on a new PR: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1452

**Comment by hypest** [comment]
> @gziolo could you please help us mergin? Since @Tug has the requested changes status, I can't press the merge button.
>
> His feedback has been taken into account and all the remaining feedback will be handled on a new PR: wordpress-mobile/gutenberg-mobile#1452

Allow me to pick this up, will admin-merge myself.

**Pull request merged**
